File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(arr, target) {
    const map = new Map();
    for(let i = 0 ; i < arr.length;i++){
        if(map.has(target-arr[i])){
            return [map.get(target-arr[i]),i]
        }else{
            map.set(arr[i],i);
        }
    }
   return []
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] arr, int target) {
        int n = arr.length;
        int[] result = new int[2];
        Map<Integer,Integer> map = new HashMap<>();
        for( int i = 0 ; i < n; i++){
            if(map.containsKey(target-arr[i])){
                result[0] = map.get(target-arr[i]);
                result[1] = i;
            }else{
                map.put(arr[i],i);
            }
        }
        return result;
    }
}